specs/0117-email-opt-in-unsubscribe.md

```diff
@@ -0,0 +1,97 @@
+- Feature Name: Email consent opt-in and Unsubscribe page
+- Created: 2024-02-21
+- Associated Issue: #3109
+
+# Summary
+
+Work required to enable email communication with Pontoon users: opt-in consent user settings, methods for collecting user opt-in, and an unsubscribe page via a single page without logging in.
+
+# Motivation
+
+There does not exist currently an email communication consent opt-in for users who register with Pontoon. This restricts our ability to reach contributors outside of the platform and reduces our effectiveness in keeping localizers informed and engaged. So we need to create settings that allow users to control their ability to opt in to email messages.
+
+This will allow us to reach out within legal compliance to the wider audience of registered users for things such as surveys, online events, etc. Future major platform enhancements will be able to build off this work, and starting this early will provide enough time to ensure interested and active contributors are opted-in ahead of time.
+
+Value for users / problems solved:
+- Users can select their preference for email communication
+- Engagement/reach expanded beyond logged-in active Pontoon users
+
+
+# Feature explanation
+
+## Email communication user settings
+
+The email opt-in consent option is available under an “Email communications” option in user preferences. “Contact email address” should be moved from Personal information to under this section and an option for non-transactional mass emails opt-ing should be added.
+
+These user email communication preferences are stored and referenced in database to determine if emails should be sent to the user.
+
+
+(Section: content)
+- Section title: “Email communications”
+- Input field: Contact email address
+- Subtitle of input: “If provided, this email address will be used for email communications and will appear under your Profile page instead of the email used for login.”
+- Check box: “Receive updates related to localization at Mozilla, projects on Pontoon, and updates to the Pontoon platform.”
+- Consent: “I’m okay with Mozilla handling my personal information as explained in this <a href="https://www.mozilla.org/en-US/privacy/websites/">Privacy Notice</a>."
+
+## Standalone page for email consent shown after existing user log-in
+
+We need to make existing users aware of the new email communication preference.
+
+Upon login (after a user’s previous session expires) an email consent page should be displayed to existing users logging in for the first time after the consent feature is enabled. 
+
+They should be shown two buttons, one to opt-in to emails or one declining to opt-in. This should not be shown again if the user clicks either of the buttons, but should be shown again upon next login if the user closed their tab/browser.
```

Comment:
```suggestion
They should be shown two buttons, one to opt in to emails or one declining to opt in. This should not be shown again if the user clicks either of the buttons, but should be shown again upon next login if the user closed their tab/browser.
```

Why do we need to wait for the next login, and not display the dialog the next time they load Pontoon?

Reply:
This could be an implementation detail to check with @mathjazz - but if we show it to them the next time they load Pontoon - does that also mean each time I open a page in a different tab would cause me to be redirected to the consent page? 

I felt doing it on login would be less disruptive to workflows. Since users have to login at least every 2 weeks and the login process forces them out of their workflow already, asking them at that time would encourage more consideration and could potentially improve conversion. (That being said its just a hypothesis and not tested, mostly based on my own personal views.)

Reply:
That's a good point, it can become invasive and they will just dismiss it without much thought.